p2p/pex/pex_reactor.go

```diff
@@ -350,8 +366,13 @@ func (r *PEXReactor) ensurePeers() {
 				} else {
 					r.book.MarkAttempt(picked)
 				}
+				// record attempt
+				r.attemptsToDial.Store(picked.DialString(), attempt+1)
+			} else {
```

Comment:
This else is unnecessary.

Reply:
how so? there is no return or break

Reply:
You are right. I assumed you break out in the if condition.